### PR TITLE
Store comment data to and fetch comment data from Datastore

### PR DESCRIPTION
**Pull request by anna-giang**
Instead of storing comments in an `ArrayList` on the servlet, store comments _persistently_ on Datastore.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -60,7 +60,6 @@ function toggleBlogPost(buttonId) {
  */
 function fetchComments() {
   fetch('/data').then(response => response.json()).then((commentData) => {
-    console.log(commentData)
```
Comment (suprsrsly) [inline portfolio/src/main/webapp/script.js:63]: This seems unrelated?

Comment (anna-giang) [reply portfolio/src/main/webapp/script.js:63]: Yeah, it doesn't affect the main functionality I implemented in the commit. But I didn't know whether a whole new commit to remove the console.log was too excessive, so I kept it in this commit - should I have done it separately?

Comment (suprsrsly) [reply portfolio/src/main/webapp/script.js:63]: I realize it seems minor, but it's pushing it quite a bit to not split this out into a new change. I think it would be borderline acceptable to remove it in this commit if code adjacent the line being deleted was being modified in this change as well, but in this case it's a whole new file.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -30,18 +34,27 @@
...
-    String commentJson = convertToJson(comments);
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<HashMap<String, String>> commentList = new ArrayList<HashMap<String, String>>();
+
+    for (Entity entity : results.asIterable()) {
+      // Each comment and the associated data (author, etc.) will be a HashMap, which converts to JSON object
+      HashMap<String, String> comment = new HashMap<String, String>();
```
Comment (suprsrsly) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: A bit of a stylistic suggestion, so optional, but what about something like

```
 commentList.add(new HashMap<String, String>(Map.of(                        
         "commentText", (String) entity.getProperty("commentText"),              
         "commentAuthor", (String) entity.getProperty("commentAuthor"))));       
```

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (anna-giang) [reply portfolio/src/main/webapp/script.js:63]: Yep, that makes sense, I'll keep that in mind in the future.

**Review by suprsrsly — APPROVED** [review]
